fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/AsyncLoadingCache.java

```diff
@@ -41,15 +41,26 @@
     private final Cache<K, CompletableFuture<V>> cache;
     private final long refreshTimeMillis;
     private final long deadlineTimeMillis;
+    private final long maxSize;
 
     public AsyncLoadingCache(long refreshTimeMillis) {
         this(refreshTimeMillis, 5000L);
     }
 
     public AsyncLoadingCache(long refreshTimeMillis, long deadlineTimeMillis) {
+        this(refreshTimeMillis, deadlineTimeMillis, Long.MAX_VALUE);
+    }
+
+    public AsyncLoadingCache(long refreshTimeMillis, long deadlineTimeMillis, long maxSize) {
         this.refreshTimeMillis = refreshTimeMillis;
         this.deadlineTimeMillis = deadlineTimeMillis;
-        cache = CacheBuilder.newBuilder().expireAfterWrite(this.refreshTimeMillis, TimeUnit.MILLISECONDS).build();
+        this.maxSize = maxSize;
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder()
+                .expireAfterWrite(this.refreshTimeMillis, TimeUnit.MILLISECONDS);
+        if (maxSize < Long.MAX_VALUE) {
```

Comment:
`UNLIMITED` is helpful, and I'm not against excluding it. Though, I guess my question here is, is there any harm in simply doing this:

` cacheBuilder.maximumSize(maxSize);`

instead of this:

```
if (maxSize != UNLIMITED) {
             cacheBuilder.maximumSize(maxSize);
}
```

Like, does it pre-allocate memory or something, that you need to keep it unset? Eyeballed `LocalCache`'s constructor, but didn't find a reason for it.

Reply:
Oh, sorry, I meant to respond earlier to this.

No, I don't think there's a, like, memory pre-allocation difference or anything. However, we follow this pattern a few other places in our code base, and it I guess means it's not doing any, like, max-size checking as things get updated, but that's probably not that big a deal either way.

Reply:
Got it. thanks.